### PR TITLE
KAFKA-2622: Add Time logical type for Copycat.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@guozhangwang Quick follow up to the logical types patch. I realized we can't avoid adding a Time type for now since we need it for SQL types in the JDBC connector. Should be a quick review since this is basically the same as the Date type, just zeroing out the date fields instead of the time fields.

**Comment by ewencp** [comment]
Actually, hold off on review. I realized there's an issue with validating values when you use Struct.put() that I still need to resolve.

**Comment by ewencp** [comment]
Ok, updated to handle logical types in validation, which is really only happening in Struct.put() for now. Should really be ready for review now, @guozhangwang

**Comment by guozhangwang** [comment]
LGTM.

File: copycat/api/src/main/java/org/apache/kafka/copycat/data/Time.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     * @return the encoded value
+     */
+    public static int fromLogical(Schema schema, java.util.Date value) {
+        if (schema.name() == null || !(schema.name().equals(LOGICAL_NAME)))
+            throw new DataException("Requested conversion of Time object but the schema does not match.");
+        Calendar calendar = Calendar.getInstance(UTC);
+        calendar.setTime(value);
+        long unixMillis = calendar.getTimeInMillis();
+        if (unixMillis < 0 || unixMillis > MILLIS_PER_DAY) {
+            throw new DataException("Copycat Time type should not have any time fields set to non-zero values.");
```
Comment (Ishiihara) [inline copycat/api/src/main/java/org/apache/kafka/copycat/data/Time.java:65]: Is the claim in the exception correct? We should have have non-zero values for some fields.

Comment (guozhangwang) [inline copycat/api/src/main/java/org/apache/kafka/copycat/data/Time.java:65]: Right. It should be "any date field"

Comment (ewencp) [inline copycat/api/src/main/java/org/apache/kafka/copycat/data/Time.java:65]: Yup, good catch. Filed https://github.com/apache/kafka/pull/291.

**Pull request closed without merging**